lib/global_uid/base.rb

```diff
@@ -164,10 +174,21 @@ def self.get_connections(options = {})
       with_connections {}
     end
 
+    def self.validate_increment!(new_record_id, connection)
+      server = self.servers.find { |s| connection.current_database.include?(s[:name]) }
+      server[:allocated_ids] << new_record_id
+
+      valid_steps = server[:allocated_ids].each_cons(2).all? do |first_identifier, second_identifier|
+        (second_identifier - first_identifier) % self.global_uid_options[:increment_by] == 0
+      end
+      raise InvalidIncrementException unless valid_steps
+    end
+
     def self.get_uid_for_class(klass, options = {})
       with_connections do |connection|
         Timeout.timeout(self.global_uid_options[:query_timeout], TimeoutException) do
```

Comment:
Separate issue that we should come back for later: `Timeout.timeout` is bad mojo and we should do the query timeout some other way

Reply:
@ggrossman perhaps we should have the alloc server connections set the `wait_timeout` MySQL session variable to the configured `self.global_uid_options[:query_timeout]`

That will remove the need for us to manage a timeout here, and the bad mojo that comes with it.

E.g. in the database.yml

```yml
test_id_server_1:
  <<: *MYSQL
  database: global_uid_test_id_server_1
  variables:
    auto_increment_increment: 5
    auto_increment_offset: 1
    wait_timeout: 10

test_id_server_2:
  <<: *MYSQL
  database: global_uid_test_id_server_2
  variables:
    auto_increment_increment: 5
    auto_increment_offset: 2
    wait_timeout: 10
```

That avoids the need for this gem to mess around with connection variables too, giving the client complete control.

Reply:
That's a good idea. I'm in favor of that. We should document that the alloc server connections should definitely declare an appropriate `wait_timeout`.

Reply:
I think it could wait for another PR but it's a good idea to get rid of `Timeout.timeout`. It can cause entirely unpredictable badness if it actually triggers.